File: app/src/main/java/com/carlos/grabredenvelope/util/LogUtil.java
package com.carlos.grabredenvelope.util;

import android.util.Log;

public class LogUtil {

    private static final String TAG = "LogUtil";

    public static void i(String msg) {
        Log.i(TAG, getLogTitle() + msg);
    }

    public static void i(String msg, Throwable throwable) {
        Log.i(TAG, getLogTitle() + msg, throwable);
    }

    public static void d(String msg) {
        Log.d(TAG, getLogTitle() + msg);
    }

    public static void d(String msg, Throwable throwable) {
        Log.d(TAG, getLogTitle() + msg, throwable);
    }

    public static void w(String msg) {
        Log.w(TAG, getLogTitle() + msg);
    }

    public static void w(String msg, Throwable throwable) {
        Log.w(TAG, getLogTitle() + msg, throwable);
    }

    public static void e(String msg) {
        Log.e(TAG, getLogTitle() + msg);
    }

    public static void e(String msg, Throwable throwable) {
        Log.e(TAG, getLogTitle() + msg, throwable);
    }

    public static String getLogTitle() {
        StringBuilder stringBuilder = new StringBuilder();
        StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
        int index = stackTraceElements.length > 4 ? 4 : stackTraceElements.length - 1;
        stringBuilder.append("[");
        stringBuilder.append(stackTraceElements[index].getClassName()).append(".");
        stringBuilder.append(stackTraceElements[index].getMethodName()).append("()").append(":");
        stringBuilder.append("lineNumber=").append(stackTraceElements[index].getLineNumber());
        stringBuilder.append("]");
        return stringBuilder.toString();
    }
}

File: app/src/main/java/com/carlos/grabredenvelope/util/LogUtil.kt
package com.carlos.grabredenvelope.util

import android.util.Log

object LogUtil {

    private val TAG = "LogUtil"

    val logTitle: String
        get() {
            val stringBuilder = StringBuilder()
            val stackTraceElements = Thread.currentThread().stackTrace
            val index = if (stackTraceElements.size > 4) 4 else stackTraceElements.size - 1
            stringBuilder.append("[")
            stringBuilder.append(stackTraceElements[index].className).append(".")
            stringBuilder.append(stackTraceElements[index].methodName).append("()").append(":")
            stringBuilder.append("lineNumber=").append(stackTraceElements[index].lineNumber)
            stringBuilder.append("]")
            return stringBuilder.toString()
        }

    fun i(msg: String) {
        Log.i(TAG, logTitle + msg)
    }

    fun i(msg: String, throwable: Throwable) {
        Log.i(TAG, logTitle + msg, throwable)
    }

    fun d(msg: String) {
        Log.d(TAG, logTitle + msg)
    }

    fun d(msg: String, throwable: Throwable) {
        Log.d(TAG, logTitle + msg, throwable)
    }

    fun w(msg: String) {
        Log.w(TAG, logTitle + msg)
    }

    fun w(msg: String, throwable: Throwable) {
        Log.w(TAG, logTitle + msg, throwable)
    }

    fun e(msg: String) {
        Log.e(TAG, logTitle + msg)
    }

    fun e(msg: String, throwable: Throwable) {
        Log.e(TAG, logTitle + msg, throwable)
    }
}
